airbyte-integrations/connectors/source-prestashop/source_prestashop/source.py

```diff
@@ -2,155 +2,47 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
+
+"""
+This file provides the necessary constructs to interpret a provided declarative YAML configuration file into
+source connector.
+
+WARNING: Do not modify this file.
+"""
+
+import logging
 import re
-from base64 import b64encode
-from typing import Any, List, Mapping, Tuple
+from typing import Any, Iterator, List, Mapping, MutableMapping, Union
 
-from airbyte_cdk.models import SyncMode
-from airbyte_cdk.sources import AbstractSource
-from airbyte_cdk.sources.streams import Stream
-from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.models import AirbyteCatalog, AirbyteConnectionStatus, AirbyteMessage, AirbyteStateMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
 
-from .streams import (
-    Addresses,
-    Carriers,
-    CartRules,
-    Carts,
-    Categories,
-    Combinations,
-    Configurations,
-    Contacts,
-    ContentManagementSystem,
-    Countries,
-    Currencies,
-    CustomerMessages,
-    Customers,
-    CustomerThreads,
-    Deliveries,
-    Employees,
-    Groups,
-    Guests,
-    ImageTypes,
-    Languages,
-    Manufacturers,
-    Messages,
-    OrderCarriers,
-    OrderDetails,
-    OrderHistories,
-    OrderInvoices,
-    OrderPayments,
-    Orders,
-    OrderSlip,
-    OrderStates,
-    PriceRanges,
-    ProductCustomizationFields,
-    ProductFeatures,
-    ProductFeatureValues,
-    ProductOptions,
-    ProductOptionValues,
-    Products,
-    ProductSuppliers,
-    ShopGroups,
-    Shops,
-    ShopUrls,
-    SpecificPriceRules,
-    SpecificPrices,
-    States,
-    StockAvailables,
-    StockMovementReasons,
-    StockMovements,
-    Stores,
-    Suppliers,
-    Tags,
-    Taxes,
-    TaxRuleGroups,
-    TaxRules,
-    TranslatedConfigurations,
-    WeightRanges,
-    Zones,
-)
 
+# Declarative Source
+class SourcePrestashop(YamlDeclarativeSource):
+    def __init__(self):
+        super().__init__(**{"path_to_yaml": "prestashop.yaml"})
 
-class SourcePrestaShop(AbstractSource):
     def _validate_and_transform(self, config: Mapping[str, Any]):
         if not config.get("_allow_http"):
```

Comment:
can we move in to spec.json? (define a regexp there?)

Reply:
I have already changed a little bit, but main idea still the same

```
if not config.get("_allow_http"):
            if not config["url"].lower().startswith("https://"):
                raise Exception(f"Invalid url: {config['url']}, only https scheme is allowed")
```

@davydov-d any idea how to move this logic on spec level ?

Reply:
what do we need `_allow_http` for?

Reply:
we still need "http" for testing